drivers/kw2xrf/kw2xrf.c

```diff
@@ -0,0 +1,1041 @@
+    reg |= MKW2XDM_PHY_CTRL4_CCATYPE(1);
+    kw2xrf_write_dreg(MKW2XDM_PHY_CTRL4, reg);
+    DEBUG("kw2xrf: Initialized and set to channel %i and pan %i.\n",
+          KW2XRF_DEFAULT_CHANNEL, KW2XRF_DEFAULT_PANID);
+
+    kw2xrf_set_option(dev, KW2XRF_OPT_AUTOACK, true);
+    kw2xrf_set_option(dev, KW2XRF_OPT_CSMA, true);
+
+    /* Switch to Receive state per default after initialization */
+    _set_sequence(dev, XCVSEQ_RECEIVE);
+    return 0;
+}
+
+int _add_cb(ng_netdev_t *dev, ng_netdev_event_cb_t cb)
+{
+    if (dev == NULL) {
+        return -ENODEV;
+    }
+    if (dev->event_cb != NULL) {
+        return -ENOBUFS;
+    }
+    dev->event_cb = cb;
+    return 0;
+}
+
+int _rem_cb(ng_netdev_t *dev, ng_netdev_event_cb_t cb)
+{
+    if (dev == NULL) {
+        return -ENODEV;
+    }
+    if (dev->event_cb != cb) {
+        return -ENOENT;
+    }
+    dev->event_cb = NULL;
+    return 0;
+}
+
+uint64_t _get_addr_long(kw2xrf_t *dev)
+{
+    uint64_t addr;
+    uint8_t *ap = (uint8_t *)(&addr);
+    for (int i = 0; i < 8; i++) {
+        ap[i] = dev->addr_long[7 - i];
+    }
+    return addr;
+}
+
+int _get(ng_netdev_t *netdev, ng_netconf_opt_t opt, void *value, size_t max_len)
+{
+    kw2xrf_t *dev = (kw2xrf_t *)netdev;
+    if (dev == NULL) {
+        return -ENODEV;
+    }
+
+    switch (opt) {
+        case NETCONF_OPT_ADDRESS:
+            if (max_len < sizeof(uint16_t)) {
+                return -EOVERFLOW;
+            }
+
+            *((uint16_t*)value) = ((dev->addr_short[0] << 8) | dev->addr_short[1]);
+            return sizeof(uint16_t);
+
+        case NETCONF_OPT_ADDRESS_LONG:
+            if (max_len < sizeof(uint64_t)) {
+                return -EOVERFLOW;
+            }
+            *((uint64_t*)value) = _get_addr_long(dev);
+            return sizeof(uint64_t);
+
+        case NETCONF_OPT_ADDR_LEN:
+            if (max_len < sizeof(uint16_t)) {
+                return -EOVERFLOW;
+            }
+            *((uint16_t *)value) = 2;
+            return sizeof(uint16_t);
+
+        case NETCONF_OPT_SRC_LEN:
+            if (max_len < sizeof(uint16_t)) {
+                return -EOVERFLOW;
+            }
+            if (dev->option & KW2XRF_OPT_SRC_ADDR_LONG) {
+                *((uint16_t *)value) = 8;
+            }
+            else {
+                *((uint16_t *)value) = 2;
+            }
+            return sizeof(uint16_t);
+
+        case NETCONF_OPT_NID:
+            if (max_len < sizeof(uint16_t)) {
+                return -EOVERFLOW;
+            }
+            *((uint16_t *)value) = dev->radio_pan;
+            return sizeof(uint16_t);
+
+        case NETCONF_OPT_CHANNEL:
+            return _get_channel(dev, (uint8_t *)value, max_len);
+
+        case NETCONF_OPT_PROTO:
+            return _get_proto(dev, (uint8_t *)value, max_len);
+
+        case NETCONF_OPT_STATE:
+            if (max_len < sizeof(ng_netconf_state_t)) {
+                return -EOVERFLOW;
+            }
```

Comment:
But this one ```NETCONF_OPT_STATE``` can remain as is? Since this is no bool; ```ng_netconf_state_t``` has 5 states or so. But for the option setters I agree and remove the checks.